### ISSUE TITLE
Allowing assigning a security policy to `twingate_group`

### ISSUE THREAD

**Issue by alexmensch** [Twingate/terraform-provider-twingate#257]
**Modified functionality**

Arguments

- (Optional) `security_policy_id` defines which security policy applies to this group. The security policy ID can be obtained from the `twingate_security_policy` and `twingate_security_policies` data sources.

Behavior

- This argument is optional because the product behavior assigns a fixed default policy to all new groups when a specific policy is not assigned. This behavior is already reflected in the existing public API.

Example usage

```terraform
data "twingate_security_policy" "devops" {
  name = "DevOps Policy"
}

resource "twingate_group" "devops" {
  name = "DevOps"
  security_policy_id = twingate_security_policy.devops.id
}
```

**Comment by wyardley** [comment]
Is there a reason the policies themselves only have a data source, but not a terraform resource?

**Comment by alexmensch** [comment]
> Is there a reason the policies themselves only have a data source, but not a terraform resource?

Yes, we don't currently provide any way to create and configure policies programmatically via our API, which then flows through to Terraform capabilities.

There are a couple reasons for this:
1. Policy configuration is quite involved, and we've made quite a few changes and updates over the last 18 months. We'd like a bit more stability before we introduce a full set of capabilities into the API.
2. The vast majority of our tenants have fewer than 5 policies, so, for now we believe the right approach is to manually configure desired policies and then choose from them programmatically. (By the way, this is our intent: we've designed our policy structure so that a small number of policies should suffice for most situations.)

We're going to be making more updates to our policy engine over the course of this year, some of which will lend themselves much better to programmatic definition.
